### PR TITLE
Add remove queued channel button

### PR DESCRIPTION
**Pull request by 0xBEEFCAF3**

### CODE REVIEW

**Comment by 0xBEEFCAF3** [comment]
Follow up PR will include a get recommended channels and reset all queued channels button

File: static/index.html [hunk 1]
```diff
@@ -76,6 +83,17 @@ <h2>PayJoin here to open these channels</h2>
...
 			}
+
+			function remove_channel(e) {
+				const table = document.getElementById("channels");
+				if (table.children.length <= 2) {
+					return
+				}
+				// Remove last two inputs
+				table.removeChild(table.children[table.children.length - 1])
+				table.removeChild(table.children[table.children.length - 2])
```
Comment (DanGould) [inline static/index.html:94]: maybe magic number are a sign the rows belong in 1 element so we can do `channels.removeChild(channels.lastChild);`

File: static/index.html [hunk 2]
```diff
@@ -44,11 +44,18 @@ <h2>Queue batches of lightning channels to open in a single transaction</h2>
...
+					<x-grid columns=2>
 						<label id="destinationLabel" span=1>Destination Node</label><label id="capacityLabel" span=1>Channel Capacity (sats)</label>
-						<input type="text" name="channels[0][node]" aria-labelledby="destinationLabel" placeholder="node@host:port" span=1 required><input type="number" id="input_sats" name="channels[0][amount]" min="20000" step="1" aria-labelledby="capacityLabel" span=1 required><!-- LND enforces minimum 20k sats channel -->
-						<button type="button" style="background: none;" span="2" onclick="add_channel()" order=1>[+]</button>
 					</x-grid>
+					<x-grid columns=2 id="channels">
+						<input type="text" name="channels[0][node]" aria-labelledby="destinationLabel" placeholder="node@host:port" span=1 required>
+						<!-- LND enforces minimum 20k sats channel -->
+						<input type="number" id="input_sats" name="channels[0][amount]" min="20000" step="1" aria-labelledby="capacityLabel" span=1 required>
+					</x-grid>
```
[the remark this answers is not in the document]
Comment (0xBEEFCAF3) [reply static/index.html]: Reverted. Added class names to ref which elements to remove

File: static/index.html [hunk 3]
```diff
@@ -76,6 +83,17 @@ <h2>PayJoin here to open these channels</h2>
 				table.appendChild(div.firstChild);
 				table.appendChild(div.lastChild)
 			}
+
+			function remove_channel(e) {
+				const table = document.getElementById("channels");
+				if (table.children.length <= 2) {
+					return
+				}
```
[the remark this answers is not in the document]
Comment (0xBEEFCAF3) [reply static/index.html]: I think the UX is a bit off if the button becomes disabled after a click. Ideally we want to have it disabled on init render. I think we can address this in a later PR.

**Comment by 0xBEEFCAF3** [comment]
@DanGould Can you take another pass at this PR. Would like to get it merged soon

**Comment by DanGould** [comment]
I made the js stateless & fixed button content

File: static/index.html (hunk 1 above)
Comment (DanGould) [reply static/index.html]: your original approach was the right one after all

**Comment by DanGould** [comment]
The test was looping infinitely on `Attempting to contact btcrpc: JSON-RPC error: transport error: Couldn't connect to host: Connection refused (os error 111)`

that's a timeout issue afaik, not an issue with this pr

**Review by DanGould — APPROVED** [review]

**Comment by DanGould** [comment]
@0xBEEFCAF3 approval to merge?

**Comment by 0xBEEFCAF3** [comment]
> @0xBEEFCAF3 do we have your approval to merge?

I don't :(

**Comment by DanGould** [comment]
you don't approve of the changes???

**Comment by 0xBEEFCAF3** [comment]
I thought you asked if I can merge. Which I can't. Feel free to merge when you think it's ready.

**Pull request merged**